shell/plugins/steve/getters.js

```diff
@@ -211,4 +213,46 @@ export default {
     return cache.generation;
   },
 
+  /**
+   * Checks the norman or steve schema resourceFields for the given path
+   */
+  pathExistsInSchema: (state, getters) => (type, path) => {
+    const schema = getters.schemaFor(type);
+
+    if (schema.requiresResourceFields && !schema.hasResourceFields) {
+      console.warn(`pathExistsInSchema requires schema ${ schema.id } to have resources fields via schema definition but none were found`); // eslint-disable-line no-console
+
+      return false;
```

Comment:
Probably a difference of opinion but in a scenario like this I'd much rather throw an exception instead of 'lie' to the consumer of this method. We can't possibly know if the schema exists unless it's loaded so how can we say it's not here?

If you wish to fail quietly here I think it might be useful to mention needing to invoke `fetchResourceFields` to remedy this issue either in a comment or in the warning (if we don't mind exposing to the browser).

Reply:
Added section to warning